text/0000-mergeable-rustdoc-cross-crate-info.md

```diff
@@ -0,0 +1,394 @@
+│       │   │   ├── sidebar-items.js
+│       │   │   └── struct.Struct.html
+│       │   ├── search.desc
+│       │   │   └── struct-crate
+│       │   │       └── struct-crate-desc-0-.js
+│       │   ├── settings.html
+│       │   └── src
+│       │       └── struct-crate
+│       │           └── lib.rs.html
+│       ├── doc.parts
+│       │   └── struct-crate
+│       │       └── crate-info
+│       └── libs.rmeta
+└── trait-crate
+    ├── src
+    │   └── lib.rs
+    └── target
+        ├── doc
+        │   ├── help.html
+        │   ├── .lock
+        │   ├── search.desc
+        │   │   └── trait-crate
+        │   │       └── trait-crate-desc-0-.js
+        │   ├── settings.html
+        │   ├── src
+        │   │   └── trait-crate
+        │   │       └── lib.rs.html
+        │   └── trait-crate
+        │       ├── all.html
+        │       ├── index.html
+        │       ├── sidebar-items.js
+        │       └── trait.Trait.html
+        ├── doc.parts
+        │   └── trait-crate
+        │       └── crate-info
+        └── libt.rmeta
+</pre>
+
+</details>
+
+## Suggested workflows
+
+With this proposal, there are three modes of invoking rustdoc: `--merge=shared`, `--merge=none`, and `--merge=finalize`.
+
+### Default workflow: mutate shared directory: `--merge=shared`
+
+In this workflow, we document a single crate, or a collection of crates into a shared output directory that is continuously updated.
+Files in this output directory are modified by multiple rustdoc invocations. Use `--merge=shared`, and specify the same `--out-dir` to every invocation of rustdoc. `--merge=shared` will be the default value if `--merge` is not provided. This is the workflow that Cargo uses, and only mode of invoking rustdoc before this RFC. This RFC is intended to enable the future deprecation of this mode.
+
+### Document crates, delaying generation of cross-crate information: `--merge=none`
+
+Document crates using a dedicated HTML output directory and a dedicated "parts" output directory. No cross-crate data nor rendered HTML output is included from other crates.
+
+This mode only renders the HTML item documentation for the current crate. It does not produce a search index, cross-crate trait implementations, or an index page. It is expected that users follow this mode with 'Link documentation' if these cross-crate features are desired.
+
+In this mode, a user will provide `--parts-out-dir=<path to crate-specific directory>` and `--merge=none` to each crate's rustdoc invocation. The user should provide `--extern-html-root-url`, and specify a absolute final destination for the docs, as a URL. The `--extern-html-root-url` flag should be provided for each crate's rustdoc invocation, for every dependency.
+
+The same `--out-dir` may be used for multiple parallel rustdoc invocations, as rustdoc will continue to acquire an flock on the `--out-dir` to address conflicts. A user may select a different `--out-dir` for each crate's rustdoc invocation. 
```

Comment:
Clarification: Why support this use-case? Is there any time that a user may want to have multiple rustdoc invocations with the same `--out-dir` while also having `--merge=none`